File: benchmarks/fair/sieve.js
const n = Number(process.env.BENCH_N) > 0 ? Number(process.env.BENCH_N) : 5000000;
const f = new Int32Array(n + 1);
let c = 0;
for (let i = 2; i <= n; i++) { if (f[i] === 0) { c++; for (let k = i * i; k <= n; k += i) f[k] = 1; } }
console.log(String(c));

File: benchmarks/fair/sieve.rs
fn main(){
    let n:usize=std::env::var("BENCH_N").ok().and_then(|v|v.parse().ok()).filter(|&v|v>0).unwrap_or(5000000);
    let mut f=vec![0i32;n+1]; let mut c:i64=0;
    let mut i=2usize;
    while i<=n { if f[i]==0 { c+=1; let mut k=i*i; while k<=n { f[k]=1; k+=i; } } i+=1; }
    println!("{}",c);
}
